hazelcast-sql/src/test/java/com/hazelcast/sql/PortableQueryTest.java

```diff
@@ -0,0 +1,231 @@
+
+/**
+ * Verifies that the member can handle queries on
+ * PortableGenericRecords when it does not have the necessary
+ * PortableFactory in its config.
+ */
+@RunWith(Parameterized.class)
+@Parameterized.UseParametersRunnerFactory(HazelcastParallelParametersRunnerFactory.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class PortableQueryTest extends HazelcastTestSupport {
+    @Parameterized.Parameter
+    public InMemoryFormat inMemoryFormat;
+
+    @Parameterized.Parameter(1)
+    public boolean clusterHavePortableConfig;
+
+    public TestHazelcastFactory factory = new TestHazelcastFactory();
+
+    @Parameterized.Parameters(name = "inMemoryFormat:{0}, clusterHavePortableConfig:{1}")
+    public static Collection<Object[]> parameters() {
+        return asList(new Object[][]{
+                {InMemoryFormat.BINARY, true},
+                {InMemoryFormat.BINARY, false},
+                {InMemoryFormat.OBJECT, true},
+                {InMemoryFormat.OBJECT, false}
+        });
+    }
+
+    @Before
+    public void setup() {
+        MapConfig mapConfig = new MapConfig("default");
+        mapConfig.setInMemoryFormat(inMemoryFormat);
+        Config config = smallInstanceConfig();
+        config.addMapConfig(mapConfig);
+        if (clusterHavePortableConfig) {
+            config.getSerializationConfig().addPortableFactory(1, new TestPortableFactory());
+        }
+        factory.newHazelcastInstance(config);
+    }
+
+    @After
+    public void cleanup() {
+        factory.terminateAll();
+    }
+
+    @Test
+    public void testQueryOnPrimitive() {
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getSerializationConfig().addPortableFactory(1, new TestPortableFactory());
+        HazelcastInstance client = factory.newHazelcastClient(clientConfig);
+        IMap<Integer, Object> map = client.getMap("test");
+        fillMap(map, 50, ChildPortable::new);
+
+        SqlResult rows = client.getSql().execute("SELECT * FROM test WHERE i >= 45");
+        AtomicInteger integer = new AtomicInteger(0);
+        rows.iterator().forEachRemaining(sqlRow -> integer.incrementAndGet());
+
+        assertEquals(5, integer.get());
+    }
+
+    @Test
+    public void testQueryOnObject() {
+        //To be ble to run comparison methods on objects on the server we need the classes
+        assumeTrue(clusterHavePortableConfig);
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getSerializationConfig()
+                .addPortableFactory(1, new TestPortableFactory());
+        HazelcastInstance client = factory.newHazelcastClient(clientConfig);
+        IMap<Integer, ParentPortable> map = client.getMap("test");
+        fillMap(map, 100, ParentPortable::new);
+        SqlService clientSql = client.getSql();
+
+        ChildPortable expected = new ChildPortable(10);
+        SqlResult rows = clientSql.execute("SELECT sa FROM test WHERE c = ?", expected);
```

Comment:
This pr's change only allows this usage(equality) when in memory is Object, right?

Reply:
Is comparison methods allowed in old predicate API without class registered in server side? How?

Reply:
It does not work on the old API as well. Both of them requires classes on the server if we want to check equality or make comparison.

Reply:
Then I think this change is somewhat unnecessary because in the old api it does not work as well and this use case is a pretty rare one

Reply:
This does not work for the new sql as well. Check the first line of test. I have just added more tests and this test is run only when there are classes on the server. 

This pr only make sure that comparison on primitive fields work when in memory format is Object. 
This was working for the old SQL. And in this pr, we make sure that it works for new sql as well.

Reply:
Hmm then I think this pr is useful. Nested portables are rarer than primitives. I don't know how much Object in-memory format is used though.